File: back-end/ivan-api/ivan-api/Models/VolunteerProfile.cs
using System;
using System.Collections.Generic;

namespace ivan_api.Models;

public partial class VolunteerProfile
{
    public int VolunteerId { get; set; }

    public int UserId { get; set; }

    public string? StudentId { get; set; }

    public string? University { get; set; }

    public string? Major { get; set; }

    public int? YearOfStudy { get; set; }

    public string? Motivation { get; set; }

    public string? Experience { get; set; }

    public string? Availability { get; set; }

    public int? VolunteerHours { get; set; }

    public decimal? Rating { get; set; }

    public int? RatingCount { get; set; }

    public bool? IsVerified { get; set; }

    public DateTime? VerifiedAt { get; set; }

    public int? VerifiedBy { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    // Additional properties for AI features
    public DateTime? LastActiveDate { get; set; }
    
    public int? TotalHoursVolunteered { get; set; }
    
    public string? Skills { get; set; }

    public virtual ICollection<Certificate> Certificates { get; set; } = new List<Certificate>();

    public virtual ICollection<EventRegistration> EventRegistrations { get; set; } = new List<EventRegistration>();

    public virtual ICollection<TaskAssignment> TaskAssignments { get; set; } = new List<TaskAssignment>();

    public virtual User User { get; set; } = null!;

    public virtual User? VerifiedByNavigation { get; set; }

    public virtual ICollection<VolunteerSchedule> VolunteerSchedules { get; set; } = new List<VolunteerSchedule>();

    public virtual ICollection<VolunteerSkill> VolunteerSkills { get; set; } = new List<VolunteerSkill>();
}

File: front-end/ivan-ui/src/types/volunteerProfile.ts
// Volunteer Profile API Types - Matching backend VolunteerProfileController

// Public DTOs for public endpoints
export interface PublicVolunteerDto {
  volunteerId: number;
  firstName: string;
  lastName: string;
  fullName: string;
  gender?: string;
  avatar?: string;
  province?: string;
  university?: string;
  major?: string;
  yearOfStudy?: number;
  motivation?: string;
  experience?: string;
  availability?: string;
  volunteerHours: number;
  rating: number;
  ratingCount: number;
  isVerified: boolean;
  totalHoursVolunteered: number;
  skills?: string;
  skillsList: PublicVolunteerSkillDto[];
}

export interface PublicVolunteerSkillDto {
  skillId: number;
  skillName: string;
  category?: string;
  proficiencyLevel: string;
  yearsOfExperience: number;
  description?: string;
}

export interface PublicVolunteerFiltersDto {
  search?: string;
  skillId?: number;
  university?: string;
  province?: string;
  isVerified?: boolean;
  page: number;
  size: number;
}

export interface SkillDto {
  skillId: number;
  skillName: string;
  category?: string;
  description?: string;
  isActive: boolean;
}

export interface VolunteerSkillDto {
  skillId: number;
  skillName: string;
  proficiencyLevel?: string;
  yearsOfExperience?: number;
  description?: string;
}

// Management DTOs for authenticated endpoints
export interface VolunteerProfileViewModel {
  volunteerId: number;
  userId: number;
  studentId?: string;
  university?: string;
  major?: string;
  yearOfStudy?: number;
  motivation?: string;
  experience?: string;
  availability?: string;
  volunteerHours: number;
  rating: number;
  ratingCount: number;
  isVerified: boolean;
  verifiedAt?: string;
  verifiedBy?: number;
  lastActiveDate?: string;
  totalHoursVolunteered: number;
  skills?: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
  // Navigation properties (from User)
  email?: string;
  fullName?: string;
  phoneNumber?: string;
  dateOfBirth?: string;
  gender?: string;
  avatar?: string;
  address?: string;
  // Verification info
  verifiedByName?: string;
  // Related entities
  volunteerSkills?: VolunteerSkillDto[];
}

export interface CreateVolunteerProfileDto {
  userId: number;
  studentId?: string;
  university?: string;
  major?: string;
  yearOfStudy?: number;
  motivation?: string;
  experience?: string;
  availability?: string;
  skills: VolunteerSkillDto[];
}

export interface UpdateVolunteerProfileDto {
  // UserProfile fields (Personal Information)
  firstName?: string;
  lastName?: string;
  phoneNumber?: string;
  dateOfBirth?: string;
  gender?: string;
  address?: string;
  wardCommune?: string;
  district?: string;
  province?: string;
  postalCode?: string;
  emergencyContactName?: string;
  emergencyContactPhone?: string;
  emergencyContactRelation?: string;
  avatar?: string;
  // VolunteerProfile fields (Volunteer-specific Information)
  studentId?: string;
  university?: string;
  major?: string;
  yearOfStudy?: number;
  motivation?: string;
  experience?: string;
  availability?: string;
  Skills: VolunteerSkillDto[]; // Changed from 'skills' to 'Skills' to match backend
}
